l10n_ch_scan_bvr/wizard/scan_bvr.py

```diff
@@ -345,21 +335,43 @@ def validate_bvr_string(self):
         #
         data = {}
         data['bvr_struct'] = self._get_bvr_structurated(self.bvr_string)
+        partner_bank_model = self.env['res.partner.bank']
+        partner_bank = False
         # We will now search the account linked with this BVR
         if data['bvr_struct']['domain'] == 'name':
             domain = [('acc_number', '=', data['bvr_struct']['beneficiaire'])]
+            partners_bank = partner_bank_model.search(domain)
+            partner_bank = len(partners_bank) == 1 and partners_bank[0]
         else:
-            domain = [
-                ('bvr_adherent_num', '=', data['bvr_struct']['bvrnumber'])]
-        partners_bank = self.env['res.partner.bank'].search(domain)
+            # A postal account that refers to a bank is uniquely identified
+            # by (beneficiaire + bvrnumber), at least by beneficiaire
+            domain = ('acc_number', '=', data['bvr_struct']['beneficiaire'])
+            partners_bank = partner_bank_model.search(domain)
+            if len(partners_bank) == 1:
+                if (
+                        not partners_bank[0].bvr_adherent_num or
+                        partners_bank[0].bvr_adherent_num ==
+                        data['bvr_struct']['bvrnumber']):
+                    partner_bank = partners_bank
+            elif len(partners_bank) > 1:
+                # We need to filter further by bvr_adherent_num
+                partners_bank.filtered(
+                    lambda r:
+                    r.bvr_adherent_num == data['bvr_struct']['bvrnumber'])
+                if len(partners_bank) > 1:
+                    raise UserError(
+                        _('There are more than one bank corresponding '
+                          'to the current string.\n'
+                          'Please check the banks configuration.'))
+                partner_bank = partners_bank
```

Comment:
You did, this variable is hardly readable or is it just me ? :-/

What I read was: `partner_bank = partner_bank` :eyes: